pkgs/top-level/haskell-packages.nix

```diff
@@ -57,7 +57,8 @@ in {
 
     ghc884 = callPackage ../development/compilers/ghc/8.8.4.nix {
       # aarch64 ghc865Binary gets SEGVs due to haskell#15449 or similar
-      bootPkgs = if stdenv.isAarch64 then
+      # Musl bindists do not exist for ghc 8.6.5, so we use 8.10.* for them
+      bootPkgs = if stdenv.isAarch64 || stdenv.targetPlatform.isMusl then
           packages.ghc8102BinaryMinimal
         else
           packages.ghc865Binary;
```

Comment:
I added the conditional for AArch64 to limit rebuilds at the time. Perhaps we could bootstrap everything with 8.10 in or after this PR.

Reply:
Sounds good to me, if we don't use GH 8.6.5 we probably should drop it as well since it would be quite behind 8.10.2 in terms of quality after this PR.